python/pyspark/sql/connect/dataframe.py

```diff
@@ -2036,6 +2036,8 @@ def _map_partitions(
         from pyspark.sql.connect.udf import UserDefinedFunction
 
         _validate_pandas_udf(func, evalType)
+        if isinstance(schema, str):
+            schema = cast(StructType, _parse_datatype_string(schema))
         udf_obj = UserDefinedFunction(
```

Comment:
the ddl string is always parsed in client side, previously it is parsed in `UserDefinedFunction`, now it is parsed here to able to cache it